### PR TITLE
Provide Insecure Mode for Go CLI

### PR DESCRIPTION
**Pull request by dubeejw**
- Allows for SSL certificate checking to be enabled, or disabled through a
  --insecure flag

### CODE REVIEW

**Comment by dubeejw** [comment]
@mdeuser, please review.

**Comment by mdeuser** [comment]
@dubeejw - some comments
- IMHO - Use cobra as much as possible for argument processing.  Piggybacking moving the httpClient initialization away from init time... the httpClient initialization can be contained in a single `Pre*Run` cobra command function.  See https://github.com/spf13/cobra#prerun-or-postrun-hooks.  This way, commands not needing httpClient support would not have the overhead of initializing a httpClient.
- I recommend that parseArgs() focus solely on parsing arguments and that the setup() (was init()) function should be called outside of parseArgs()
- Remove the `\n` character from the errMsg strings - the string used to create the error objects.  See commands.go line 37

**Comment by dubeejw** [comment]
@mdeuser, ready for a new review. I need to run it through the automation before it gets merged.

**Comment by dubeejw** [comment]
All tests passed.

File: tools/go-cli/go-whisk-cli/commands/commands.go [hunk 1]
```diff
@@ -24,10 +24,47 @@ import (
...
+    clientConfig := &whisk.Config{
+        AuthToken:  Properties.Auth,
+        Namespace:  Properties.Namespace,
+        BaseURL:    baseURL,
+        Version:    Properties.APIVersion,
+        Insecure:   flags.global.insecure,
+    }
+
+    // Setup client
+    client, err = whisk.NewClient(http.DefaultClient, clientConfig)
```
Comment (mdeuser) [inline tools/go-cli/go-whisk-cli/commands/commands.go]: @dubeejw - The init() method also initializes the client variable with a new NewClient object.  Is this still needed?

**Comment by dubeejw** [comment]
@mdeuser, how about now?

**Comment by csantanapr** [comment]
rebase and merged 0e92f6cace3ab8d2fe8b7f8178c4603f450a7537

**Pull request closed without merging**